FMSgridtools/make_topog/topogobj.py

```diff
@@ -12,51 +13,85 @@
 class TopogObj():
     output_name: str = None
     ntiles: int = None
-    nx: int = None
-    ny: int = None
+    nx: dict = None
+    ny: dict = None
     x_refine: int = None
     y_refine: int = None
     scale_factor: float = None
-    depth: npt.NDArray[np.float64] = None
+    depth_vars: dict = None
+    depth_vals: dict = None
     dataset: xr.Dataset = None
     __data_is_generated: bool = False
+    global_attrs: dict = None
+    dims: dict = None
 
-    # sets up dataset variables
+    # applies any scaling factors or refinements given 
     def __post_init__(self):
-        depth_vars = dict()
-        depth_attrs = dict()
-        depth_attrs["standard_name"] = "topographic depth at T-cell centers"
-        depth_attrs["units"] = "meters"
-        depth_coords = xr.Coordinates( {'ny': np.empty(self.ny), 'nx': np.empty(self.nx)} )
-        # TODO check dims of depth
+        self.depth_vars = dict()
+        self.depth_attrs = dict()
+        self.depth_attrs["standard_name"] = "topographic depth at T-cell centers"
+        self.depth_attrs["units"] = "meters"
+
+        # make sure nx/ny dicts are given
+        if self.nx is None or self.ny is None:
+            raise ValueError("No nx/ny dictionaries provided, cannot construct TopogObj")
+
+        # adjust nx/ny for refinements and scaling factor
+        if self.x_refine is not None:
+            print(f"updating nx for x refine value of {self.x_refine}")
+            self.nx.update((tname, val/self.x_refine) for tname, val in self.nx.items()) 
+        if self.y_refine is not None:
+            print(f"updating ny for y refine value of {self.y_refine}")
+            self.ny.update((tname, val/self.y_refine) for tname, val in self.ny.items()) 
+        if self.scale_factor is not None:
+            print(f"updating nx/ny for scale factor value of {self.scale_factor}")
+            self.nx.update((tname, val*self.scale_factor) for tname, val in self.nx.items()) 
+            self.ny.update((tname, val*self.scale_factor) for tname, val in self.ny.items()) 
+
+        # set up coordinates and dimensions based off tile count and nx/ny values
         # if single tile exclude tile number in variable name
-        if (self.ntiles == 1):
-            depth_vars["depth"]= xr.DataArray(
-                coords=depth_coords,
-                dims=["ny", "nx"],
-                attrs=depth_attrs,
-            )
+        if self.ntiles == 1:
+            nx_curr_tile = self.nx['tile1']
+            ny_curr_tile = self.ny['tile1']
+            self.dims = ['ny', 'nx']
```

Comment:
for realistic topog, does it read in a topog file to remap data to the grid specified in the mosaic?   do the dims need to match up with the dims in the read-in topog file?